docs/architecture/overview.md

```diff
@@ -0,0 +1,193 @@
+
+> [!IMPORTANT]
+> The app host isn't a production runtime. It's a development-time orchestration tool that simplifies the process of running and debugging your application locally.
+
+In this section, several key questions are answered to help you understand how the app host orchestrates your app model:
+
+- **What powers the orchestration?**
+
+  Orchestration is delegated to the [Microsoft Developer Control Plane](#developer-control-plane) (DCP), which manages resource lifecycles, startup order, dependencies, and network configurations across your app topology.
+
+- **How is the app model used?**
+
+  The app model defines all resources via implementations of <xref:Aspire.Hosting.ApplicationModel.IResource>, including containers, processes, databases, and external services—forming the blueprint for orchestration.
+
+- **What role does the app host play?**
+
+  The app host provides a high-level declaration of the desired application state. It delegates execution to DCP, which interprets the app model and performs orchestration accordingly.
+
+- **What resources are monitored?**
+
+  All declared resources—including containers, executables, and integrations—are monitored to ensure correct behavior and to support a reliable dev loop.
+
+- **How are containers and executables managed?**
+
+  Containers and processes are initialized with their configs, launched in order, and networked as needed. DCP ensures their readiness and connectivity during orchestration.
+
+- **How are resource dependencies handled?**
+
+  Dependencies are defined in the app model and evaluated by DCP to determine correct startup sequencing, ensuring resources are available before dependents start.
+
+- **How is networking configured?**
+
+  Networking—such as port bindings—is autoconfigured unless explicitly defined. DCP resolves conflicts and ensures availability, enabling seamless communication between services.
+
+The orchestration process follows a layered architecture. At its core, the app host represents the developer's desired view of the distributed application's resources. DCP ensures that this desired state is realized by orchestrating the resources and maintaining consistency.
+
+The [app model](../fundamentals/app-host-overview.md#define-the-app-model) serves as a blueprint for DCP to orchestrate your application. Under the hood, the app host is a .NET console application powered by the [📦 Aspire.Hosting.AppHost](https://www.nuget.org/packages/Aspire.Hosting.AppHost) NuGet package. This package includes build targets that register orchestration dependencies, enabling seamless dev-time orchestration.
+
+The .NET Aspire app host contains an implementation of the `k8s.KubernetesClient` (from the [📦 KubernetesClient](https://www.nuget.org/packages/KubernetesClient) NuGet package), which is a .NET client for Kubernetes. This client is used to communicate with the DCP API server, allowing the app host to delegate orchestration tasks to DCP.
```

Comment:
Before this I would add a sentence that explains that DCP is a Kubernetes-compatible API server, from the network protocol perspective. Otherwise it is not super clear why the app host is using K8s libraries to talk to DCP.

Reply:
(you have that later in the doc, I see, but here it comes as a surprise)